src/core/src/dimension.cpp

```diff
@@ -39,6 +40,23 @@ size_t merge_labels(const size_t label_a, const size_t label_b, bool merge_unequ
         return 0;
 }
 
+int64_t stringToInt64(const std::string& valStr) {
+    int64_t ret{0};
+    std::istringstream ss(valStr);
+    if (!ss.eof()) {
+        ss >> ret;
+    }
+    return ret;
+}
+
+bool check_all_digits(const std::string& value) {
+    auto val = ov::util::trim(value);
+    for (const auto& c : val) {
+        if (!std::isdigit(c) || c == '-')
```

Comment:
will we accept `-10` for dimension?

Reply:
No, this method checks that all characters are digits. If '-' is found it returns false.
"-1" case is handled separately outside of this method in line 89